engine/swarm/secrets.md

```diff
@@ -0,0 +1,770 @@
+
+## Examples
+
+This section includes three graduated examples which illustrate how to use
+Docker secrets. The images used in these examples have been updated to make it
+easier to use Docker secrets. To find out how to modify your own images in
+a similar way, see
+[Build support for Docker Secrets into your images](#build-support-for-docker-secrets-into-your-images).
+
+### Simple example: Get started with secrets
+
+This simple example shows how secrets work in just a few commands. For a
+real-world example, continue to
+[Use secrets with a service](secrets.md#example-use-secrets-with-a-service).
+
+1.  Add a secret to Docker. The `docker secret create` command reads standard
+    input.
+
+    ```bash
+    $ echo "This is a secret" | docker secret create my_secret_data -f -
+    ```
+
+2.  Create a `redis` service and grant it access to the secret. By default,
+    the container can access the secret at `/run/secrets/<secret_name>`, but
+    you can customize the file name on the container using the `target` option.
+
+    ```bash
+    $ docker service  create --name="redis" --secret="my_secret_data" redis:alpine
+    ```
+
+3.  Verify that the task is running without issues using `docker service ps`. If
+    everything is working, the output looks similar to this:
+
+    ```bash
+    $ docker service ps --no-trunc redis
+
+    NAME                               lIMAGE         NODE              DESIRED STATE  CURRENT STATE           ERROR  PORTS
+    redis.1.cezsyv6xljjj6bzpajbc6agkb  redis:alpine   ip-172-31-46-109  Running        Running 26 seconds ago
+    ```
+
+    If there were an error, and the task were failing and repeatedly restarting,
+    you would see something like this:
+
+    ```bash
+    $ docker service ps redis
+
+    NAME                      IMAGE         NODE  DESIRED STATE  CURRENT STATE          ERROR                      PORTS
+    redis.1.siftice35gla      redis:alpine  moby  Running        Running 4 seconds ago                             
+     \_ redis.1.whum5b7gu13e  redis:alpine  moby  Shutdown       Failed 20 seconds ago      "task: non-zero exit (1)"  
+     \_ redis.1.2s6yorvd9zow  redis:alpine  moby  Shutdown       Failed 56 seconds ago      "task: non-zero exit (1)"  
+     \_ redis.1.ulfzrcyaf6pg  redis:alpine  moby  Shutdown       Failed about a minute ago  "task: non-zero exit (1)"  
+     \_ redis.1.wrny5v4xyps6  redis:alpine  moby  Shutdown       Failed 2 minutes ago       "task: non-zero exit (1)"
+    ```
+
+4.  Get the ID of the `redis` service task container using `docker ps` , so that
+    you can use `docker exec` to connect to the container and read the contents
+    of the secret data file, which defaults to being readable by all and has the
+    same name as the name of the secret.  The ID is the first field of the
+    output of `docker ps`.
+
+    ```bash
+    $ docker ps --filter name=redis
```

Comment:
That seems needlessly complicated, and it seems useful to teach the reader about the --filter thing..